[MLIR] Vector to XeGPU conversion: Use proper source variant for create_nd_tdesc op creation.

silee2 opened the pull request:
If source strided memref is not fully static - at least one of shape, strides, offset is kDynamic - use i64 source variant.
With this change, xegpu.create_nd_tdesc created by lowering from vector dialect, can rely on getMixedOffsets, getMixedSize and getMixedStrides to get relevant values.

silee2 committed:
[MLIR] Vector to XeGPU conversion: Use proper source variant for
create_nd_tdesc op creation.
If source strided memref is not fully static - at least one of
shape, strides, offset is kDynamic - use i64 source variant.

silee2 committed:
Remove dead code.

silee2 committed:
Merge remote-tracking branch 'origin/main' into updateVectorToXeGPU

charithaintc reviewed (approved):
LGTM.

Jianhui-Li commented on mlir/lib/Conversion/VectorToXeGPU/VectorToXeGPU.cpp:
```diff
@@ -102,18 +102,47 @@ static xegpu::CreateNdDescOp createNdDescriptor(PatternRewriter &rewriter,
+  if (offset == ShapedType::kDynamic)
+    isStatic = false;
+
+  for (auto stride : strides) {
```
Just wonder whether we use ShapedType::isStatic() to check both offset/shape/strides?

silee2 replied:
Switched to using ShapedType::isStatic()

silee2 committed:
Combine base addr and byte offset as adjusted base addr.

silee2 committed:
Address reviewer comments.

silee2 committed:
Merge remote-tracking branch 'origin/main' into updateVectorToXeGPU

silee2 committed:
[MLIR] Fix broken GPU integration tests targetting SYCL and LevelZero runtime.
SPIR-V kernel lowering no longer supports i1 store.
Fuse GPU kernels to remove usage of i1 stores.

silee2 committed:
Revert "[MLIR] Fix broken GPU integration tests targetting SYCL and LevelZero runtime."

This reverts commit f43d5c8c981baabdb497ce1ca6b429c6f836a9eb.

Jianhui-Li reviewed (approved):
LGTM

silee2 committed:
Merge branch 'main' into updateVectorToXeGPU
